### PR TITLE
POWR-1756 URL paths for advisory groups, programs, projects with identical names

### PR DESCRIPTION
**Pull request by kxwang**

### CODE REVIEW

File: web/modules/custom/portland/portland.module [hunk 1]
```diff
@@ -13,10 +13,9 @@ use Drupal\Core\Cache\Cache;
 // This file concerns itself with setting up tokens with hooks
 include 'portland.tokens.inc';
 
-
-// The plural is used in the list page URL. E.g. "/bds/services"
-// The group list view's path must match the plural form shown here.
-// If the group list view doesn't exist, visiting the alias will get a 404.
+/**
```
Comment (joshuami) [inline web/modules/custom/portland/portland.module]: I like the move to block comments, but Drupal coding standards still suggest a line length of 80 characters—which makes PR review a lot easier as well. https://www.drupal.org/node/1354

File: web/modules/custom/portland/portland.module [hunk 2]
```diff
@@ -198,11 +213,8 @@ function portland_group_content_type_delete(GroupContentTypeInterface $groupCont
 
 /**
  * Implements hook_ENTITY_TYPE_presave().
+ * POWR-475 Changing short name of a group can cause the group URL to be different from the URLs in the sidebar menu. Root cause is that the group URL is set by PathAuto to be clean and unique. But the URL in sidebar menu is formed by Twig code using the short name as is. The solution is to update short name to ensure it's clean and unique when the group is updated.
```
Comment (joshuami) [inline web/modules/custom/portland/portland.module]: "sidebar menu" should be updated to "group menu". We moved that menu inline under group header information.

File: web/modules/custom/portland/portland.module [hunk 3]
```diff
@@ -213,7 +225,7 @@ function portland_group_presave(GroupInterface $group) {
     }
     // Check if this is a new group or an existing group whose shortname has been modified
     if( $originalShortname !== $newShortname) {
-        $cleanShortname = \Drupal::service("pathauto.alias_cleaner")->cleanString($newShortname);
+        $cleanShortname = $newShortname;
```
Comment (joshuami) [inline web/modules/custom/portland/portland.module]: I wonder if we want to take this chance to rename this field to group_path. I'm a little worried that all of our references to shortname don't really make sense in this new context. 🤔

File: web/modules/custom/portland/portland.module [hunk 4]
```diff
@@ -224,20 +236,51 @@ function portland_group_presave(GroupInterface $group) {
     }
 }
 
+/**
+ * A helper function to clean up group shortname.
```
Comment (joshuami) [inline web/modules/custom/portland/portland.module]: This is a slick function.

File: web/modules/custom/portland/portland.module [hunk 5]
```diff
@@ -839,4 +889,44 @@ function portland_form_alter(&$form, &$form_state, $form_id) {
             }
         }
     }
+    else if(substr_compare($form_id, $group_form_prefix, 0, strlen($group_form_prefix)) === 0) {
```
Comment (joshuami) [inline web/modules/custom/portland/portland.module]: I'm thinking this is a lot of code in the overall portland.module. Should we use this as an opportunity to move some of the group functions into a submodule? 🤔

File: web/modules/custom/portland/portland.module (hunk 1 above)
Comment (kxwang) [reply web/modules/custom/portland/portland.module]: Trimmed block comments

File: web/modules/custom/portland/portland.module (hunk 2 above)
Comment (kxwang) [reply web/modules/custom/portland/portland.module]: Updated

File: web/modules/custom/portland/portland.module (hunk 5 above)
Comment (kxwang) [reply web/modules/custom/portland/portland.module]: Probably will do that as another story. This one is getting complicated now. :)

File: web/sites/default/config/field.field.group.bureau_office.field_group_path.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+uuid: b430cde1-a011-4728-a181-7b683dd5e538
+langcode: en
+status: true
+dependencies:
+  config:
+    - field.storage.group.field_group_path
+    - group.type.bureau_office
+  module:
+    - custom_add_another
```
Comment (joshuami) [inline web/sites/default/config/field.field.group.bureau_office.field_group_path.yml]: I wonder why this was a dependency. Doesn't seem like this field would have an "add another" function. 🤔

Comment (kxwang) [reply web/sites/default/config/field.field.group.bureau_office.field_group_path.yml]: Not sure how it got in there. This is a singleton field.

**Pull request closed without merging**